### PR TITLE
(CISC-972) allow parallel provisioning of systems

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

**Comment by da-ar** [comment]
(CISC-972) Handle bulk tear_down would rely upon - https://github.com/puppetlabs/provision/pull/146/commits/1f5ce833059c387a5c5bf1240acf20fb3d0404ef

File: lib/puppet_litmus/rake_helper.rb [hunk 1]
```diff
@@ -173,6 +173,18 @@ def tear_down_nodes(targets, inventory_hash)
         next if node_name == 'litmus_localhost'
 
         result = tear_down(node_name, inventory_hash)
+        # Some provisioners tear_down targets that were created as a batch job.
+        # These provisionsers should return the list of additional targets
```
Comment (DavidS) [inline lib/puppet_litmus/rake_helper.rb:177]:
```suggestion
        # These provisioners should return the list of additional targets
```

File: lib/puppet_litmus/rake_tasks.rb [hunk 2]
```diff
@@ -375,7 +388,7 @@
             at_exit { exit! }
 
             env = options[:env].nil? ? {} : options[:env]
-            env['HONEYCOMB_TRACE'] = Honeycomb.current_span.to_trace_header
+            env['HTTP_X_HONEYCOMB_TRACE'] = Honeycomb.current_span.to_trace_header
```
Comment (DavidS) [inline lib/puppet_litmus/rake_tasks.rb:391]: please undo this change

File: lib/puppet_litmus/rake_tasks.rb [hunk 3]
```diff
@@ -393,7 +406,7 @@
           spinners = TTY::Spinner::Multi.new("[:spinner] Running against #{targets.size} targets.")
           payloads.each do |title, test, options|
             env = options[:env].nil? ? {} : options[:env]
-            env['HONEYCOMB_TRACE'] = Honeycomb.current_span.to_trace_header
+            env['HTTP_X_HONEYCOMB_TRACE'] = Honeycomb.current_span.to_trace_header
```
Comment (DavidS) [inline lib/puppet_litmus/rake_tasks.rb:409]: please undo this change

**Review by DavidS — COMMENTED** [review]
some minor sidenotes
